elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java

```diff
@@ -156,14 +151,32 @@ private void populateMetaData(MetaDataStore metaDataStore) {
                 .forEach(metaDataStore::addTable);
     }
 
+    public interface Transaction extends AutoCloseable {
+        @Override
+        void close();
+    }
+
+    public abstract Transaction beginTransaction();
+
     /**
      * Executes the specified {@link Query} against a specific persistent storage, which understand the provided
      * {@link Query}. Results may be taken from a cache, if configured.
      *
      * @param query The query customized for a particular persistent storage or storage client
+     * @param transaction
      * @return query results
      */
-    public abstract QueryResult executeQuery(Query query);
+    public abstract QueryResult executeQuery(Query query, Transaction transaction);
+
+    /**
+     * Get a serial number or other token indicating the version of the data in the table.
+     * No particular semantics are required, though it must change if the data changes.
+     * If one is not available, returns null.
```

Comment:
You might want to mention that if one is not available, returns null which disables caching.